Fix crop raster episode

rogerkuou opened the pull request:
- [x] Use `overview` as the base map #30
- [x] Convert polygon cropping from exercise to instruction
- [x] Improve the visualization of wells cropping #27 
- [x] Design an exercise for the corrupted waterway data #43
- [x] Simplify the visualization code blocks #44

rogerkuou commented:
I made some changes for Episodes 7 and 8. This mainly addresses some issues we found in the test workshop and also solving the corrupted waterway shp file issue by adding an instruction for correcting the flipped coordinates.

cpranav93 commented on _episodes/08-crop-raster-data.md:
```diff
@@ -49,17 +49,21 @@ print(true_color_image.shape)
+<img src="../fig/20-crop-raster-overview-raster-00.png" title="Overview of the raster"  width="512" style="display: block; margin: auto;" />
 
-But do we need the entire raster? Suppose we are interested in the crop fields. To compare its coverage with the raster data, we first check the coordinate systems of both raster and vector data. For raster data, we use `pyproj.CRS`:
+As we can see the overview image is much smaller comparing to the original true color image. Therefore the visualization is much faster. Suppose we are interested in the crop fields. We would like to know where are crop fields locate in the image. To compare its coverage with the raster data, we first check the coordinate systems of both raster and vector data. For raster data, we use `pyproj.CRS ...
```
```suggestion
As we can see, the overview image is much smaller compared to the original true color image. Therefore the visualization is much faster. If we are interested in the crop fields, then we would like to know where these are located in the image. To compare its coverage with the raster data, we first check the coordinate systems of both raster and vector data. For raster data, we use `pyproj.CRS`:
```

cpranav93 commented on _episodes/08-crop-raster-data.md:
```diff
@@ -129,40 +133,27 @@ from matplotlib import pyplot as plt
 the raster. Therefore before actual processing, we can first crop the raster to
 our area of interest. The `clip_box` function allows one to crop a raster by the
-min/max of the x and y coordinates.
+min/max of the x and y coordinates. Note that we are croping the original image `true_color_image` now, but not the overview image `overview_image`.
```
```suggestion
min/max of the x and y coordinates. Note that we are croping the original image `true_color_image` now, and not the overview image `overview_image`.
```

cpranav93 commented on _episodes/08-crop-raster-data.md:
```diff
@@ -183,96 +174,41 @@ raster_clip.plot.imshow(figsize=(8,8))
 ## Crop raster data with a geometry buffer
 
-It is not always the case that the AoI comes in the format of a polygon. Sometimes one would like to perform analysis around a (set of) point(s), or polyline(s). For example, in our AoI, there are also some groundwater monitoring wells coming as point vector data. One may also want to perform analysis around these wells. The location of the wells is stored in `data/groundwater_monitoring_well`.
+It is not always the case that the AoI comes in the format of polygon. Sometimes one would like to perform analysis around a (set of) point(s), or polyline(s). For example, in our AoI, there are also some groundwater monitoring wells coming as point vector data. One may also want to perform analysis around these wells. The location of the wells is stored in `data/groundwater_monitoring_well`. 
```
AoI is not defined anywhere in this lesson. Maybe we want to mention what it is the first time we use the acronym.

rogerkuou replied:
I think it was mentioned in Line 18 of this episode

> It is quite common that the raster data you have in hand is too large to process, or not all the pixels are relevant to your area of interest (AoI).

cpranav93 replied:
Oops my bad! That's fine then!

cpranav93 commented on _episodes/08-crop-raster-data.md:
```diff
@@ -183,96 +174,41 @@ raster_clip.plot.imshow(figsize=(8,8))
 ## Crop raster data with a geometry buffer
 
-It is not always the case that the AoI comes in the format of a polygon. Sometimes one would like to perform analysis around a (set of) point(s), or polyline(s). For example, in our AoI, there are also some groundwater monitoring wells coming as point vector data. One may also want to perform analysis around these wells. The location of the wells is stored in `data/groundwater_monitoring_well`.
+It is not always the case that the AoI comes in the format of polygon. Sometimes one would like to perform analysis around a (set of) point(s), or polyline(s). For example, in our AoI, there are also some groundwater monitoring wells coming as point vector data. One may also want to perform analysis around these wells. The location of the wells is stored in `data/groundwater_monitoring_well`. 
```
```suggestion
It is not always the case that the AoI comes in the format of polygon. Sometimes one would like to perform analysis around a (set of) point(s), or polyline(s). For example, in our AoI, there are also some groundwater monitoring wells available as point vector data. One may also want to perform analysis around these wells. The location of the wells is stored in `data/groundwater_monitoring_well`. 
```

rogerkuou replied:
```suggestion
It is not always the case that the Area of Interest (AoI) comes in the format of polygon. Sometimes one would like to perform analysis around a (set of) point(s), or polyline(s). For example, in our AoI, there are also some groundwater monitoring wells coming as point vector data. One may also want to perform analysis around these wells. The location of the wells is stored in `data/groundwater_monitoring_well`. 
```

cpranav93 commented on _episodes/08-crop-raster-data.md:
```diff
@@ -294,60 +234,78 @@ wells.plot(ax=ax, color='red', markersize=2)
 {: .language-python}
+<img src="../fig/20-crop-raster-well-buffers-over-raster-05.png" title="Raster croped by buffer around wells" width="512" style="display: block; margin: auto;" />
+
+The red dots grow larger, which means they are converted from point to buffer polygons.
```
```suggestion
The red dots have grown larger indicating the conversion from point to buffer polygons.
```

cpranav93 commented on _episodes/08-crop-raster-data.md:
```diff
@@ -413,8 +371,7 @@ cropped_raster.plot.imshow(figsize=(8,8))
 
```
For the command above, do we need to add an image for the visualise?

rogerkuou replied:
Thx for pointing this out! In fact, after reading this section again, I realized that I was doing the previous reprojection in another way, which is causing the two images here in the same CRS instead of different CRS as stated in the text. I updated the intro text and added a new image. Would you like to give it another look?

cpranav93 replied:
Hey, I tried to take a look at the image but I cannot see the figure. Is the file pushed?

rogerkuou replied:
You are right, I forgot to push the image. It should be there now.

cpranav93 commented on _episodes/08-crop-raster-data.md:
```diff
@@ -413,8 +371,7 @@ cropped_raster.plot.imshow(figsize=(8,8))
 >
-> This time, let's crop the `crop_fields` image using the `true_color_image` image. Discuss
-> the results.
+> This time let's do the other way around. Let's crop the `crop_fields` image using the `true_color_image` image. Discuss the results.
```
```suggestion
> This time let's do it the other way around by cropping the `crop_fields` image using the `true_color_image` image. Discuss the results.
```

cpranav93 reviewed (changes requested):
It reads great Ou and everything seems to be working well! Just a few corrections here and there that I have raised as comments/suggestions.

fnattino commented on _episodes/08-crop-raster-data.md:
```diff
@@ -294,66 +234,87 @@ wells.plot(ax=ax, color='red', markersize=2)
 crop_fields = rioxarray.open_rasterio("crop_fields.tif")
+
+# Reproject to RD
+crop_fields = crop_fields.rio.reproject("EPSG:28992")
```
I am not sure why we need to reproject the image here? When doing `retroject_match` we would anyway convert `true_color_image` to the CRS of `crop_fields`, whatever that is - right?

rogerkuou replied:
Yes. Actually this is an artificial step to put them in different CRS. The purpose of this is to show that the `retroject_match` can do reprojection and match in one step. Will add some more description text for this.

fnattino reviewed (changes requested):
Hi @rogerkuou , very nice! I have suggested some minor changes and add a couple of questions, but overall looks great to me.

rogerkuou commented on _episodes/07-vector-data-in-python.md:
```diff
@@ -248,5 +248,86 @@ different features.
+> > ![Wrong waterways](../fig/E07-waterways-wrong.png)
+> {: .solution}
+{: .challenge}
+
```
Thanks @fnattino! I actually thought that the file was corrupted. Thanks for adding this info on other conventions. I will add the part in.

rogerkuou commented:
Hi @cpranav93 and @fnattino, thanks for your review! Could please give a final check?


Just mentioning the two unresolved threads:
- @cpranav93 I added the missing image. Thanks for pointing out!
- @fnattino the reprojection you mentioned is an artificial reprojection to demonstrate the `reproject_match` function. I added some description for this.

cpranav93 commented:
Hi @rogerkuou, All good for me! Great job.
